crates/bevy_color/src/spectral.rs

```diff
@@ -0,0 +1,279 @@
+        luminance: 1.0,
+    };
+
+    /// Monochromatic light in the 589 nm wavelength, typically produced by sodium vapor lamps
+    pub const SODIUM_VAPOR: Self = Self {
+        wavelength: 589.0,
+        luminance: 1.0,
+    };
+
+    /// Create a new spectral color with the given wavelength and luminance
+    pub const fn new(wavelength: f32, luminance: f32) -> Self {
+        Self {
+            wavelength,
+            luminance,
+        }
+    }
+
+    /// Convert the spectral color to a Linear Rgba color, using the CIE 1931 2-deg XYZ color matching functions
+    pub fn to_linear(&self) -> LinearRgba {
+        if self.wavelength < Self::CIE_1931_NM_CMF_LOOKUP_TABLE_START
+            || self.wavelength >= Self::CIE_1931_NM_CMF_LOOKUP_TABLE_END
+        {
+            // If the wavelength is outside the range of the lookup table, return black
+            return LinearRgba::BLACK;
+        }
+
+        let index = ((self.wavelength - Self::CIE_1931_NM_CMF_LOOKUP_TABLE_START)
+            / Self::CIE_1931_NM_CMF_LOOKUP_TABLE_INCREMENT)
+            .floor() as usize;
+
+        let lerp = (self.wavelength - Self::CIE_1931_NM_CMF_LOOKUP_TABLE_START)
+            % Self::CIE_1931_NM_CMF_LOOKUP_TABLE_INCREMENT
+            / Self::CIE_1931_NM_CMF_LOOKUP_TABLE_INCREMENT;
+
+        let row = Self::CIE_1931_XYZ_CMF_LOOKUP_TABLE[index];
+        let next_row = Self::CIE_1931_XYZ_CMF_LOOKUP_TABLE[index + 1];
+
+        let x = row[0].lerp(next_row[0], lerp);
+        let y = row[1].lerp(next_row[1], lerp);
+        let z = row[2].lerp(next_row[2], lerp);
+
+        let xyza = Xyza::new(x, y, z, 1.0);
+
+        let mut linear = Color::from(xyza).to_linear();
+
+        // Clamp negative values to zero
+        linear.red = linear.red.max(0.0);
+        linear.green = linear.green.max(0.0);
+        linear.blue = linear.blue.max(0.0);
+
+        // Apply luminance scaling, without clamping to white
+        (linear * self.luminance).with_alpha(1.0)
+    }
+
+    /// Returns a new spectral color with the given wavelength
+    pub fn with_wavelength(&self, wavelength: f32) -> Self {
+        Self {
+            wavelength,
+            ..*self
+        }
+    }
+
+    /// The wavelength where the look up table starts
+    const CIE_1931_NM_CMF_LOOKUP_TABLE_START: f32 = 355.0;
+
+    /// The wavelength where the look up table ends
+    const CIE_1931_NM_CMF_LOOKUP_TABLE_END: f32 = 835.0;
+
+    /// The increment between each row in the look up table
+    const CIE_1931_NM_CMF_LOOKUP_TABLE_INCREMENT: f32 = 5.0;
+
+    /// CIE 1931 2-deg, XYZ color matching functions, in lookup table form.
+    /// Each row is a 5nm step from 360nm to 830nm (inclusive), with two
+    /// rows of sentinel values, at the start and end of the table.
+    /// (For interpolation to zero.)
+    ///
+    /// Source: <http://cvrl.ioo.ucl.ac.uk/plotcmfs.php>
+    #[allow(clippy::excessive_precision)]
+    const CIE_1931_XYZ_CMF_LOOKUP_TABLE: [[f32; 3]; 97] = [
```

Comment:
I would recommend to use the approximations https://github.com/cadacoon/polylight/blob/main/src/color.rs#L88

Reply:
That link is not working for me 😕 Maybe it's a private repo, or they removed it?